tokio-util/src/task/join_queue.rs

```diff
@@ -183,6 +184,55 @@ impl<T> JoinQueue<T> {
         std::future::poll_fn(|cx| self.poll_join_next_with_id(cx)).await
     }
 
+    /// Tries to join the next task in FIFO order if it has completed.
+    ///
+    /// Returns `None` if the queue is empty or if the next task is not yet ready.
+    pub fn try_join_next(&mut self) -> Option<Result<T, JoinError>> {
+        let jh = self.0.front_mut()?;
+        if jh.is_finished() {
+            // Since this function is not async and cannot be forced to yield, we should
+            // disable budgeting when we want to check for the `JoinHandle` readiness.
+            let jh = tokio::task::coop::unconstrained(jh);
+            let res = jh.now_or_never()?;
```

Comment:
Emm, why should we return `None` here?

Reply:
Do you suggest me to return back `jh.now_or_never().unwrap()` here with a comment about jh completeness?

Reply:
> I'm not entirely sure if this scenario exists: when jh has finished and cooperative scheduling is disabled, jh.poll still returns a Pending.

Could we consider such scenario as bug in `tokio` + `tokio-util` combo? In this case panic is more suitable, I can even make something like `.expect("Tokio bug happened, please open an issue")`.